src/query/functions/binary/binary.go

```diff
@@ -72,7 +71,7 @@ func processBinary(
 
 		// NB(arnikola): this is a sanity check, as scalar comparisons
 		// should have previously errored out during the parsing step
-		if !params.ReturnBool {
+		if !params.ReturnBool && isComparison {
```

Comment:
shouldn't we be leaving this to the parser ? Seems like duplication of logic.

Reply:
Agreed, that's why it's specifically a sanity check that we don't get into weird situations here